clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -9948,7 +9948,8 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
                 ",+SPV_INTEL_hw_thread_queries"
                 ",+SPV_KHR_uniform_group_instructions"
                 ",+SPV_INTEL_masked_gather_scatter"
-                ",+SPV_INTEL_tensor_float32_conversion";
+                ",+SPV_INTEL_tensor_float32_conversion"
+                ",+SPV_INTEL_optnone";
```

Comment:
Does it matter that this extension is being set regardless of any optimization level check?

Reply:
My test of -O2 w/ and w/o this PR show generated spirv files are the same.

In some cases O0 device code could be linked with O2 device code. We still need to preserve optnone if they are only present in some functions.